### ISSUE TITLE
[FEATURE] - Keep SysMain enabled as it's likely useful for responsiveness

### ISSUE THREAD

**Issue by Evonos** [Atlas-OS/Atlas#1093]
### What is your feature request regarding to?

Atlas Playbook

### Is your feature request related to a problem? Please describe.

No issue ( if you dont consider slower loading and making the OS less "smart" at managing ram not an issue )

### Describe the solution you would like.

Sysmain as Optional setting to be enabled during install , or maybe even enable it anyway.

### Describe alternatives you have considered.

None , its kinda a smart feature any OS does ( Ios , Linux , Android and windows with it enabled too )

### Additional context.

Sysmain is "Prefetching" data into empty ram , theres no latency or performance issues because windows leaves allways a certain % Zeroed ( Entirely empty )  even on a NVME with 2 gb DRAM prefetching is still beneficial Example screenshots why Prefetching is beneficial ( Screens of Ram map ) after boot. like of my 32gb RAM 17 are in use , 14 in Standby and roughly 1gb entirely free ( Zeroed ) 
![2024-05-01_14-52](https://github.com/Atlas-OS/Atlas/assets/23293584/a75f2092-6792-4105-a1c9-0c919e4a4524)

a Likely much better read for than my text already hugely explained
https://new.reddit.com/r/computers/comments/c8iq9o/dont_disable_sysmain_previously_known_as/

Also Empty ram is  wasted ram and Atlas OS shouldnt disable things ( or maybe make a switch for Max disabling and Preferred disabling) which Improves performance / quality of the OS

Luckily you can simply enable the Sysmain service and its already set to "3" in registry and starts working.

**Comment by he3als** [comment]
I agree. Unless there's a proven problem with SuperFetch on SSDs, I don't think it should be disabled. Microsoft also doesn't recommend the service to be disabled for desktop users.

**Comment by Xyueta** [comment]
SysMain will be enabled by default and left for the user's choice in the Atlas Configuration folder since the next release.

> Okay maybe actually buggy , Sysmain is running and prefetch set to 3 , yet it stopped working and preloading stuff into ram weirdly. not sure why yet maybe a task was disabled or something related to sysmain or a needed service.

Just to note, SysMain is a very poorly-described option/feature of Windows, so it is hard to research it.

Thanks.

**Comment by he3als** [comment]
I'm going to re-open this as I think it's best to keep issues open until there's a new release for transparency with users.

**Comment by NoelJacob** [comment]
How to temporarily enable it before installing the latest playbook?

**Comment by he3als** [comment]
https://raw.githubusercontent.com/Atlas-OS/Atlas/main/src/playbook/Executables/AtlasDesktop/3.%20General%20Configuration/Superfetch/Enable%20SuperFetch.cmd

**Comment by Evonos** [comment]
> https://raw.githubusercontent.com/Atlas-OS/Atlas/main/src/playbook/Executables/AtlasDesktop/3.%20General%20Configuration/Superfetch/Enable%20SuperFetch%20(default).cmd

that link is broken now. "404 not found"

**Comment by Ast3risk-ops** [comment]
> > https://raw.githubusercontent.com/Atlas-OS/Atlas/main/src/playbook/Executables/AtlasDesktop/3.%20General%20Configuration/Superfetch/Enable%20SuperFetch%20(default).cmd
> 
> 
> 
> that link is broken now. "404 not found"

We did some reshuffling and the file is gone, you can find it using the commit history.

**Comment by Xyueta** [comment]
> > https://raw.githubusercontent.com/Atlas-OS/Atlas/main/src/playbook/Executables/AtlasDesktop/3.%20General%20Configuration/Superfetch/Enable%20SuperFetch%20(default).cmd
> 
> that link is broken now. "404 not found"

here: 
https://github.com/Atlas-OS/Atlas/tree/main/src/playbook/Executables/AtlasDesktop/6.%20Advanced%20Configuration/Services/Superfetch

**Comment by Evonos** [comment]
> https://raw.githubusercontent.com/Atlas-OS/Atlas/main/src/playbook/Executables/AtlasDesktop/3.%20General%20Configuration/Superfetch/Enable%20SuperFetch%20(default).cmd

Another issue.
This script doesnt fix the entry "BaseTime" in "Computer\HKEY_LOCAL_MACHINE\SYSTEM\CurrentControlSet\Control\Session Manager\Memory Management\PrefetchParameters"

which Prevents Prefetch from working correctly.
the entry is "2b1de5a0" but should be "0" or 1 ( but 1 is worse )

heres an Explanation
BaseTime Entry:
The “BaseTime” entry is a DWORD value (32-bit) in the registry.
By default, its value is set to 0.

When set to 1, it modifies the behavior of the Prefetch cache.
Behavior with Value 0 (Default):
When “BaseTime” is set to 0, Windows uses the current timestamp as the base for Prefetch calculations.
This means the Prefetch cache optimizes based on recent program and file executions.

Behavior with Value 1:
When “BaseTime” is set to 1, Windows uses a fixed timestamp as the base for Prefetch calculations.
This may result in less dynamic Prefetch behavior, potentially not aligning perfectly with your usage patterns.


i dont know why "2b1de5a0"  is here but likely it was there because of the disabling of Prefetch before from Atlas OS,
so the script should be Adjusted to Reset "Basetime" back to "0" for best functionality ( and this should also be considered on next Atlas Release ) 

i dont know if "BootId" also needs an Adjustment ill observe it.

**Comment by he3als** [comment]
If you want to re-enable SysMain:
1. Go to https://raw.githubusercontent.com/Atlas-OS/Atlas/main/src/playbook/Executables/AtlasDesktop/6.%20Advanced%20Configuration/Services/Superfetch/Enable%20SuperFetch%20(default).cmd
2. On the page, right-click and click 'Save as'
3. If not already a `.cmd` file on download, rename it in File Explorer to have a `.cmd` extension
4. Run the script
5. Restart your computer

**Comment by Protator906** [comment]
Pls, should you decide to re-enable SysMain in 0.5, and I hope you wont, at least make it an option, not the default. 
Its absence is part of why 0.4 runs so well. 

"[..] as it's likely useful for responsiveness"
Negatory, ghostrider. The feature was outdated by years already when MS implemented it. 
Your computer has an SSD boot drive? Then you don't need it.
All it does is collect user data about which software you run and when, under the pretense of 'prefetching' data on time to speed up program launch. People have discussed the feature's usefulness and tested their system with it enabled and disabled to the moon and back. 
It's useless on a system less than 15 years old. 
SysMain's other features are just as amazing. Like memory compression. It can be useful on systems with very little ram, like an old thinclient with 4GB. Otherwise it's useless and just another feature that wastes clockcycles. Because it's broken. 
Memory management as a whole has been broken for ages. My workstations all have between 64 and 256GB of RAM. 
Most of the time 90+% of the memory is free and available. Nothing allocated for stuff in the background etc.. 
Yet somehow SysMain thinks it's a good idea to compress part of the memory. 
I doubt that needlessly compressing and decompressing data will increase a system's responsiveness. 
It's on the same level as W7/10/11's management of virtual memory. The more physical memory you have installed, the bigger the automatically generated pagefile. People have been complaining about that nonsensical default behavior for two decades now ... have they bothered to fix it? Nope. 
So to assume a feature is actually useful simply because MS says so ... ^^

**Comment by he3als** [comment]
Fixed in [release v0.4.1](https://github.com/Atlas-OS/Atlas/releases/tag/0.4.1). See the release for more info on what v0.4 users can do.

**Comment by respberryx** [comment]
The idea that enabling SysMain is a must for performance improvements is just not true, especially in modern setups. SysMain's job is to "preload" frequently used apps into RAM, but this was meant for systems running on HDDs. With SSDs, where access speeds are already extremely fast, this prefetching barely adds anything and can sometimes create more problems than it solves.

There are cases where "SysMain causes high disk and CPU usage," which leads to performance hits rather than improvements, especially on systems with SSDs. Microsoft itself know that there are issues with SysMain, stating that it may "cause high CPU usage on 64-bit apps"​. If you read some more, even Microsoft said that some apps that used /LARGEADDRESSAWARE:NO while compiling will make CPU usage to spike up to 100% every 1-2 minutes in-definitely. High disk usage spikes due to SysMain are also a well-documented issue, especially with systems that have older hardware or limited resources​.

Claiming "empty RAM is wasted RAM" oversimplifies modern memory management. Windows doesn’t just leave RAM empty; it manages what’s in RAM based on actual usage patterns. Not every system benefits from SysMain's behavior, and disabling it can often lead to a more responsive experience, particularly on faster storage setups. @Xyueta
